ql/src/java/org/apache/hadoop/hive/ql/lockmgr/DbTxnManager.java

```diff
@@ -485,6 +480,26 @@ private void clearLocksAndHB() {
     stopHeartbeat();
   }
 
+  private void cleanupDirForCTAS() {
```

Comment:
could we pass a `Context ctx` here instead of initializing the destination table when acquiring locks?

Reply:
Currently `Context ctx` is not available during `rollbackTxn` which is why I chose to store the object.
However I agree passing `Context ctx` is better.
There are multiple ways of doing this - 
First would be to involve passing `Context ctx` to `rollbackTxn` method which would change the HiveTxnManager API itself (I particularly dont like this since this would be a breaking change).

Or we could create a new function in the `HiveTxnManager` interface of the same name and call it from the driver when rollback conditions are satisfied.

My idea was to avoid both and initialise the destination in one of the existing APIs but I am open for any other suggestions.

Reply:
As you mentioned, we need to add an overloaded `rollbackTxn` method into the `HiveTxnManager` interface that accepts `Context ctx` param.

Reply:
Done